daemon/api.go

```diff
@@ -367,6 +374,59 @@ func webify(result map[string]interface{}, resource string) map[string]interface
 	return result
 }
 
+func searchStore(c *Command, r *http.Request) Response {
+	route := c.d.router.Get(snapCmd.Path)
+	if route == nil {
+		return InternalError("router can't find route for snaps")
+	}
+
+	query := r.URL.Query()
+
+	auther, err := c.d.auther(r)
+	if err != nil && err != auth.ErrInvalidAuth {
+		return InternalError("%v", err)
+	}
+
+	remoteRepo := newRemoteRepo()
+	found, err := remoteRepo.FindSnaps(query.Get("q"), query.Get("channel"), auther)
+	if err != nil {
+		status := http.StatusInternalServerError
+		if e, ok := err.(*store.HTTPError); ok && e != nil {
+			status = e.StatusCode
+		}
+
+		return &resp{
+			Type: ResponseTypeError,
+			Result: &errorResult{
+				Message: err.Error(),
+			},
+			Status: status,
+		}
+	}
+
+	meta := &Meta{
+		SuggestedCurrency: remoteRepo.SuggestedCurrency(),
+	}
+
+	results := make([]*json.RawMessage, len(found))
+	for i, x := range found {
+		resource := ""
+		url, err := route.URL("name", x.Name())
+		if err == nil {
```

Comment:
that's a pretty bogus snap though, then if the point is not to die on bad data we should skip the entire result, not just have an empty resource